open-apparel-registry/open-apparel-registry
# Add script for running one-off management commands on ECS

## Overview

Add an `ecsmanage` management command to allow running one-off Django management commands as ECS tasks on staging or production.

Closes #167.

## Demo

<img width="843" alt="screen shot 2019-02-12 at 12 27 35 pm" src="https://user-images.githubusercontent.com/14170650/52655198-9da65400-2ec1-11e9-824b-e6af663bab43.png">

## Notes

* We had discussed a few possible approaches to this command, including pure shell AWS CLI and Boto. The command-line parsing wound up being complex enough that I went with a Django management command using Boto for the sake of simplicity.

* The `-e|--env` flag controls the environment, so we can run e.g. production migrations like so:

```
$ ./scripts/manage ecsmanage --env production migrate
```

## Testing Instructions

* Shell into the VM
* Run `./scripts/manage ecsmanage help`, the equivalent of `python manage.py help`
  * Note that this will run on the staging environment by default
* Click on the link to the details page and confirm that it takes you to the right place
* On the details page, navigate to `Containers > django > Log Configuration`, open up the logs, and confirm that the command exited successfully
